crates/compiler/load_internal/tests/test_load.rs

```diff
@@ -2131,3 +2131,51 @@ fn roc_file_no_extension() {
 
     assert_eq!(err, expected, "\n{}", err);
 }
+
+#[test]
+fn roc_package_depends_on_other_package() {
+    let modules = vec![
+        (
+            "main",
+            indoc!(
+                r#"
+            package [Module] { other: "other/main.roc" }
+            "#
+            ),
+        ),
+        (
+            "Module.roc",
+            indoc!(
+                r#"
+            module [foo]
+
+            import other.OtherMod
+
+            foo = OtherMod.say "hello"
+            "#
+            ),
+        ),
+        (
+            "other/main.roc",
+            indoc!(
+                r#"
+            package [OtherMod] { other: "other/main.roc" }
```

Comment:
This will probably not work in the future since I don't think we want to allow cyclical package dependencies, including the trivial cycle of self dependency. If possible, could you make a third package and have the second depend on it?